src/docs/arch-session.dox

```diff
@@ -0,0 +1,53 @@
+/*! @arch_page arch-session Session
+
+@section arch_session_def Definition
+After a @ref arch-connection has been established between the application and WiredTiger, the
+application can start sending requests to WiredTiger using a session. A session is internally
+represented by WT_SESSION and plays an important role since almost all operations require one.
+
+A session can only be created through an existing connection with the API
+WT_CONNECTION::open_session. It is possible to create multiple sessions through the same connection.
+In fact, one connection can have multiple sessions but one session can only be associated to one
+connection. The maximum number of sessions is set through the configuration item \c session_max as
+part of the configuration string given in ::wiredtiger_open.
+
+Sessions that are created by the calling application are called "user session". It exists another
+type of session called "internal session". Internal sessions are created and managed by WiredTiger
```

Comment:
I feel like we need to rephrase this sentence. The previous sentence ends with "internal session". This one starts with the same words and in the middle again "WiredTiger internally". Maybe we can start this sentence with `This type of sessions are created...`?

Reply:
@AlexeyAnisimovAU  The section has been reworked in 40e6075, the word "internal" is still used a few times but it does not disturb me that much. What do you think ?